tools/src/main/java/org/apache/kafka/tools/consumer/group/ShareGroupCommand.java

```diff
@@ -366,6 +378,67 @@ Entry<Throwable, Map<String, Throwable>> sendDeleteShareGroupOffsetsRequest(Stri
             return new SimpleImmutableEntry<>(topLevelException, topicLevelResult);
         }
 
+        void resetOffsets() throws ExecutionException, InterruptedException {
+            String groupId = opts.options.valueOf(opts.groupOpt);
+            Map<TopicPartition, OffsetAndMetadata> offsetsToReset = prepareOffsetsToReset(groupId);
+            if (offsetsToReset == null) {
+                return;
+            }
+            boolean dryRun = opts.options.has(opts.dryRunOpt) || !opts.options.has(opts.executeOpt);
+            if (!dryRun) {
+                adminClient.alterShareGroupOffsets(groupId,
+                    offsetsToReset.entrySet().stream()
+                        .collect(Collectors.toMap(
+                            Entry::getKey,
+                            entry -> entry.getValue().offset()
+                        ))
+                ).all().get();
+            }
+            Set<SharePartitionOffsetInformation> partitionOffsets = mapOffsetsToSharePartitionInformation(groupId, offsetsToReset);
+            Map<String, ShareGroupDescription> shareGroups = describeShareGroups(Collections.singleton(groupId));
+            TreeMap<String, Entry<ShareGroupDescription, Collection<SharePartitionOffsetInformation>>> groupOffsets = new TreeMap<>();
+            shareGroups.forEach((id, description) -> groupOffsets.put(id, new SimpleImmutableEntry<>(description, partitionOffsets)));
+            printOffsets(groupOffsets, opts.options.has(opts.verboseOpt));
```

Comment:
The `--verbose` option is not valid with `--reset-offsets` so this is a bit weird. But actually, I think it would be better if there was something like the consumer group tool `printOffsetsToReset`. This is just because I think having a column heading `NEW-START-OFFSET` would make it clear that it's the new start offset.